incubator/hnc/pkg/reconcilers/hierarchical_namespace.go

```diff
@@ -30,20 +33,72 @@ type HierarchicalNamespaceReconciler struct {
 }
 
 // Reconcile sets up some basic variables and then calls the business logic.
-// It currently only generates some logs for testing.
+// It currently handles basic creation of namespace and hierarchyconfig.
 func (r *HierarchicalNamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// TODO report error state if the webhook is bypassed - see issue
 	//  https://github.com/kubernetes-sigs/multi-tenancy/issues/459
 	if ex[req.Namespace] {
 		return ctrl.Result{}, nil
 	}
 
+	ctx := context.Background()
 	log := r.Log.WithValues("trigger", req.NamespacedName)
 	log.Info("Reconciling HNS")
 
+	nm := req.Name
+	pnm := req.Namespace
+
+	// We want to trigger the hierarchyConfig reconciliation on the self-serve namespace, since
+	// the parent's 'requiredChildren' field will be deprecated. The HierarchyConfig Reconciler
+	// will do the real job of updating the hierarchy in the forest and creating namespaces and
+	// hierarchyconfig (hc) instances.
+	//
+	// Just enqueuing an hc instance in the memory for hc reconciliation will lose its parent
+	// information since the hc reconciler will get an empty instance from the apiserver.
```

Comment:
Maybe specify what "its" mean here? Does it mean the current namespace?

Reply:
SG, the comment is updated!

It means the parent field in the hc object of the current self-serve subnamespace.